src/Symfony/Component/DomCrawler/CrawlerInterface.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DomCrawler;
+
+/**
+ * @extends \IteratorAggregate<int, \DOMNode>|\IteratorAggregate<int, \DOM\Node>
+ */
+interface CrawlerInterface extends \Countable, \IteratorAggregate
```

Comment:
the Crawler class has a lot more public methods than the one you added in the interface. Changing the return type of methods to this interface instead of the Crawler class would thus mean that IDE and static analysis tool would restrict you to the methods of this interface instead of the full API of the component. This looks quite bad to me.

Reply:
I fulfilled the interface. Does it look better?